### ISSUE TITLE
Task warning messages should be recorded in the suite run-db.

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#1490]
(currently they're not).

**Comment by matthewrmshin** [comment]
@hjoliver I am currently in the middle of adding some new tables and doing some refactor to `cylc.rundb`. Please let me know the kind of things you would like to add.

**Comment by hjoliver** [comment]
@matthewrmshin - are you working on #864? I'll attempt to put a bit of thought into this tomorrow...

**Comment by matthewrmshin** [comment]
Not really. I am mainly working on #992 (and by extension #1052).

**Comment by benfitzpatrick** [comment]
E.g. `cylc message -p WARN 'Oh no!'` to be recorded in the suite database.

**Comment by hjoliver** [comment]
Non-output task messages can be sent with NORMAL, WARNING, or CRITICAL priority.  This might seem a more-or-less useless feature.  Thinking back, the intention was to log all messages in case of typos that caused output messages not to be recognized as outputs, and to allow deliberate recording of task errors centrally back when task job logs were not so easily accessed.   There is probably little need for this now (although the first reason is still valid) but given that we do support such messages they should really be recorded in the DB as well as logged.  Easy to do - the "message" column in the task_states table is already used for message outputs.

Changing the Issue title to "all messages" as there's no reason to single out WARNINGs...
